### PR TITLE
Implement configurable hotkey feature

### LINKED ISSUE #5: UI: Ability to set hotkey for toggling Vietnamese mode on/off

### PR DESCRIPTION
**Pull request by huytd**
This PR added some preliminary works for #5 

1. Implement a hotkey matcher: Now we can create a hotkey definition from a key sequence string like `"super + ctrl + space"`, `"shift + Z"`, and use this hotkey to toggle the Vietnamese input mode
2. Render the configured hotkey to the UI: translate the configured hotkey into a proper symbols, like <kbd>⌃</kbd> <kbd>⇧</kbd> <kbd>Space</kbd>. This implementation also handle the different symbol between the platforms, for example <kbd>❖</kbd> for super key on Linux, and <kbd>⌘</kbd> on macOS.

hotkey-screenshot

Fixes #5 #6

### CODE REVIEW

File: src/hotkey.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    pub fn is_match(&self, modifiers: KeyModifier, keycode: &char) -> bool {
+        return self.modifiers == modifiers && self.keycode.eq_ignore_ascii_case(keycode);
+    }
+}
+
+impl Display for Hotkey {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut keys = String::new();
+        if self.modifiers.is_control() {
+            keys.push_str(SYMBOL_CTRL);
```
Comment (unrealhoang) [inline src/hotkey.rs:45]: write!(f, "{} ", SYMBOL_CTRL)?;

**Review by unrealhoang — APPROVED** [review]

File: src/hotkey.rs (hunk 1 above)
Comment (huytd) [reply src/hotkey.rs]: ✅ updated

**Review by unrealhoang — APPROVED** [review]

File: src/hotkey.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    pub fn is_match(&self, modifiers: KeyModifier, keycode: &char) -> bool {
+        return self.modifiers == modifiers && self.keycode.eq_ignore_ascii_case(keycode);
+    }
+}
+
+impl Display for Hotkey {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.modifiers.is_control() {
+            _ = write!(f, "{} ", SYMBOL_CTRL);
```
Comment (unrealhoang) [inline src/hotkey.rs:44]: why not ```write!(f, "{} ", SYMBOL_CTRL)?;``` to bubble up error?

Comment (huytd) [reply src/hotkey.rs:44]: oh u're right

Comment (huytd) [reply src/hotkey.rs]: ✅ updated

File: src/hotkey.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            write!(f, "{} ", SYMBOL_ALT)?;
+        }
+        if self.modifiers.is_super() {
+            write!(f, "{} ", SYMBOL_SUPER)?;
+        }
+        write!(
+            f,
+            "{}",
+            match self.keycode {
+                KEY_ENTER => "Enter".to_owned(),
```
Comment (unrealhoang) [inline src/hotkey.rs:59]: ```write!(f, "Enter")?``` instead

Comment (huytd) [reply src/hotkey.rs]: ✅

**Review by ZeroX-DG — APPROVED** [review]
LGTM 🎉

**Comment by huytd** [comment]
Last batch of implementation. There are a lot of BS in this last commit, but I'm getting tired of this, so let it be.
Will refactor later on, now, it works.

image

**Review by unrealhoang — APPROVED** [review]

**Pull request merged**